builder/azure/arm/step_get_ip_address.go

```diff
@@ -11,52 +11,86 @@ import (
 	"github.com/mitchellh/packer/packer"
 )
 
+type NetworkCommunicationType int
```

Comment:
You're right.  Network is probably the wrong term.  Endpoint may be closer, but I am not sure it accurately captures the variance across public, VIP, DIP, etc.

I will go with Endpoint, and we can change later if necessary.